### ISSUE TITLE
False Positive - "Unnecessary braces" on one-line functions

### ISSUE THREAD

**Issue by timlagrande** [rust-lang/rust-analyzer#15790]
rust-analyzer version v0.3.1697
rustc 1.71.0 (8ede3aae2 2023-07-12)

Although stylistically inappropriate, single line functions such as 
"pub async fn not_found_handler() -> impl IntoResponse { StatusCode::NOT_FOUND }"
generate the following warning: 

warning: unnecessary braces around block return value
  --> src/routes/base_routes.rs:63:55
   |
63 | pub async fn not_found_handler() -> impl IntoResponse {StatusCode::NOT_FOUND}
   |                                                       ^                     ^
   |
   = note: `#[warn(unused_braces)]` on by default
help: remove these braces
   |
63 - pub async fn not_found_handler() -> impl IntoResponse {StatusCode::NOT_FOUND}
63 + pub async fn not_found_handler() -> impl IntoResponse StatusCode::NOT_FOUND
   |

Obviously implementing this suggestion results in many more warnings

**Comment by lnicola** [comment]
Can't reproduce using the code below. Can you test it or provide a minimal example?

```rust
use axum::response::IntoResponse;
use http::StatusCode;

pub async fn not_found_handler() -> impl IntoResponse { StatusCode::NOT_FOUND }
```

And that looks like a `rustc` warning, ours only works in `use` statements. I couldn't reproduce it on 1.71 either, so please check if it shows up with `cargo clean; cargo check`.
